indexing-service/src/main/java/io/druid/indexing/overlord/TaskQueue.java

```diff
@@ -499,14 +506,25 @@ private void syncFromStorage()
     try {
       if (active) {
         final List<Task> newTasks = taskStorage.getActiveTasks();
+        List<Task> addedTasks = Lists.newArrayList(Sets.difference(Sets.newHashSet(newTasks), Sets.newHashSet(tasks)));
```

Comment:
Also probably need to do difference on taskId instead of task object here as well, unless we make sure each Task has a proper hash and equals.